bib/cmd/bootc-image-builder/image.go

```diff
@@ -50,6 +50,11 @@ type ManifestConfig struct {
 func Manifest(c *ManifestConfig) (*manifest.Manifest, error) {
 	rng := createRand()
 
+	// this is currently mostly here so that tests need less setup
```

Comment:
I suggest just returning an error here. We can just pass the architecture from the test suite, no need to be smart here, I think this can bite us later.

Reply:
Sure, I removed it and started https://github.com/osbuild/bootc-image-builder/pull/365

Reply:
Fwiw I think this will just crash "naturally" if c.Architecture is not set, this probably fixes it in a nicer way:
```go
--- a/bib/cmd/bootc-image-builder/image.go
+++ b/bib/cmd/bootc-image-builder/image.go
@@ -103,6 +103,8 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
                                QCOW2Compat: "1.1",
                        },
                }
+       default:
+               return nil, fmt.Errorf("Unknown architecture %q", c.Architecture)
        }
 
        if kopts := customizations.GetKernel(); kopts != nil && kopts.Append != "" {
@@ -302,6 +304,8 @@ func manifestForISO(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest, erro
                        },
                        UEFIVendor: "fedora",
                }
+       default:
+               return nil, fmt.Errorf("Unknown architecture %q", c.Architecture)
        }
 
        img.OSName = "default"
```
which probably should also be de-duplicated.